scripts/jsondeps.gradle

```
@@ -127,33 +127,53 @@ allprojects {
                 return "\"${config.name}\":[${combined}]"
             }
 
+            // Gets dependencies of resolved configurations.
+            // If the configuration is not resolvable, or exception is thrown returns 0 dependencies.
+            // 
+            // Note:
+            // -----
+            // Some configurations aren't meant to be resolved, because they're just meant to be containers of
+            // dependency constraints. This only occurs for gradle version v3.3 onwards.
+            // 
+            // References:
+            // -----------
+            // - https://discuss.gradle.org/t/what-is-a-configuration-which-cant-be-directly-resolved/30721
+            // - https://docs.gradle.org/current/userguide/declaring_dependencies.html#sec:resolvable-consumable-configs
+            def getResolvedConfigDepsOrEmptyOf = { config -> 
+                def configsDeps = null              
+                try {
+                    // We know that we are working against gradle version greater or equal to v3.3
+                    // So, configuration has resolvable property. If the configuration is not resolvable return null.
+                    if (config.respondsTo("isCanBeResolved")) {
+                        if (config.isCanBeResolved() == false) {
+                            println "DEBUG: Configuration is not resolvable"
+                            return null
+                        }
+                    }
+
+                    // At this point, we are working against gradle: 
+                    // - v3.3 or greater, and configuration is resolvable.
+                    // - v3.2.1 or lower, and configuration does not have canResolve property and are inherently resolvable.
```

(Sorry to delete your previous comment @meghfossa - GitHub code review otherwise has an irritating bug that does not let me create a multiline comment that ends on a previously commented line.)

Rewording suggestion for clarity:

```suggestion
                    // At this point, we know one of these must be true: 
                    // - We are on Gradle v3.3 or newer (because isCanBeResolved is present) and the configuration is resolvable (isCanBeResolved() == true, therefore did not exit early above).
                    // - We are on Gradle v3.2.1 or older (because isCanBeResolved is not present). Configurations in Gradle v3.2.1 or older are always resolvable.
```